src/libraries/System.Net.Security/src/System/Net/Security/SslStream.IO.cs

```diff
@@ -111,30 +111,32 @@ private Task ProcessAuthenticationAsync(bool isAsync = false, CancellationToken
         {
             ThrowIfExceptional();
 
-            Activity? activity = s_activitySource.StartActivity(ActivityName, IsServer ? ActivityKind.Server : ActivityKind.Client);
-            try
+            if (NetSecurityTelemetry.Log.IsEnabled() || s_activitySource.HasListeners())
             {
-                if (NetSecurityTelemetry.Log.IsEnabled())
-                {
-                    return ProcessAuthenticationWithTelemetryAsync(isAsync, cancellationToken);
-                }
-                else
-                {
-                    return isAsync ?
-                        ForceAuthenticationAsync<AsyncReadWriteAdapter>(IsServer, null, cancellationToken) :
-                        ForceAuthenticationAsync<SyncReadWriteAdapter>(IsServer, null, cancellationToken);
-                }
+                return ProcessAuthenticationWithTelemetryAsync(isAsync, cancellationToken);
             }
-            finally
+            else
             {
-                activity?.Stop();
+                return isAsync ?
+                    ForceAuthenticationAsync<AsyncReadWriteAdapter>(IsServer, null, cancellationToken) :
+                    ForceAuthenticationAsync<SyncReadWriteAdapter>(IsServer, null, cancellationToken);
             }
         }
 
         private async Task ProcessAuthenticationWithTelemetryAsync(bool isAsync, CancellationToken cancellationToken)
         {
-            NetSecurityTelemetry.Log.HandshakeStart(IsServer, _sslAuthenticationOptions.TargetHost);
-            long startingTimestamp = Stopwatch.GetTimestamp();
+            long startingTimestamp;
+            if (NetSecurityTelemetry.Log.IsEnabled())
+            {
+                NetSecurityTelemetry.Log.HandshakeStart(IsServer, _sslAuthenticationOptions.TargetHost);
+                startingTimestamp = Stopwatch.GetTimestamp();
+            }
+            else
+            {
+                startingTimestamp = 0;
+            }
+
+            Activity? activity = s_activitySource.StartActivity(ActivityName, IsServer ? ActivityKind.Server : ActivityKind.Client);
```

Comment:
nit: you don't need the `finally` block. you can do `using Activity? activity = s_activitySource.StartActivity(ActivityName, IsServer ? ActivityKind.Server : ActivityKind.Client);` which will automatically stop the activity when exiting the block.